File: Task1110/Sources/Task1110/validation.swift
import Foundation
#if DEVELOPING
import TestData

let data1 = TestData(name: "Data 1", text: """
4 3
1135
""", expected: """
5
""")

let data2 = TestData(name: "Data 2", text: """
7 10
1475842
""", expected: """
0
""")

let data3 = TestData(name: "Data 3", text: """
33 2
314159265358979323846264338327950
""", expected: """
647094440
""")

let data4 = TestData(name: "Data 4", text: """
""", expected: """
""")

let data5 = TestData(name: "Data 5", text: """
""", expected: """
""")

let validationList = [
    data1,
    data2,
    data3,
//    data4,
//    data5,
]

// ----------------------------------------------------------
func print(_ items: Any...) {
    TestData.write(items)
}

func validate(_ main: (() -> String) -> Void) {
    validationList.forEach { $0.execute(main) }
}
// ----------------------------------------------------------

#endif

File: Task729/src/Validation.kt
import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream
import java.io.PrintStream

fun main() {
    val data1 = TestData("Data 1", """
        2 3
""".trimIndent(), """
        6
""".trimIndent())

    val data2 = TestData("Data 2", """
        1 1
""".trimIndent(), """
        1
""".trimIndent())

    val data3 = TestData("Data 2", """        
""".trimIndent(), """
""".trimIndent())

    val data4 = TestData("Data 2", """        
""".trimIndent(), """
""".trimIndent())

    val data5 = TestData("Data 2", """        
""".trimIndent(), """
""".trimIndent())

    val validationList = listOf(
        data1,
        data2,
//        data3,
//        data4,
//        data5,
    )
    
    println()
    validationList.forEach { it.execute(::solve) }
    println("--------------------------------")
    println()
}

class TestData(
    private val name: String,
    private val data: String,
    private val expected: String,
) {
    fun execute(block: () -> Unit) {
        println("--- [$name] -------------------")
        val originalIn = System.`in`
        val originalOut = System.out
        val actual = ByteArrayInputStream(data.toByteArray()).use { inputStream ->
            ByteArrayOutputStream().use { outputStream ->
                System.setIn(inputStream)
                try {
                    System.setOut(PrintStream(outputStream))
                    try {
                        block()
                        outputStream.toString()
                    } finally {
                        System.setOut(originalOut)
                    }
                } finally {
                    System.setIn(originalIn)
                }
            }
        }
        print(actual)

        if (expected.isNotEmpty()) {        
            if (actual.trim() == expected.trim()) {
                println("✅ OK")
            } else {
                println("❌ NG")
            }
        }
    }
}
